packages/neuron-ui/src/widgets/Calendar/utils.ts

```diff
@@ -0,0 +1,115 @@
+import { useTranslation } from 'react-i18next'
+
+export interface Day {
+  instance: Date
+  year: number
+  month: number
+  date: number
+  weekday: number
+  curMonth: boolean
+  isToday: boolean
+  label: string
+  selectable: boolean
+}
+
+interface DateRange {
+  minDate: Date | null
+  maxDate: Date | null
+}
+
+export function dayInRange(date: Date, range: DateRange) {
+  if (range.minDate !== null) {
+    range.minDate.setHours(0, 0, 0, 0)
+    if (date < range.minDate) {
+      return false
+    }
+  }
+  if (range.maxDate !== null) {
+    range.maxDate.setHours(0, 0, 0, 0)
+    if (date > range.maxDate) {
+      return false
+    }
+  }
+  return true
+}
+
+export function monthInRange(year: number, month: number, range: DateRange) {
+  return dayInRange(new Date(year, month + 1, 1), range)
+}
+
+export function yearInRange(year: number, range: DateRange) {
+  return dayInRange(new Date(year + 1, 1, 1), range)
```

Comment:
I think we can judge by `getFullYear`. Otherwise: `yearInRange(2022, { minDate: new Date(2023, 0, 1), maxDate: new Date(2023, 0, 31) } )` will be false.
And I don't know why we should increase the year with 1

Reply:
really a bug, my fault